ionoscloud/resource_lan.go

```diff
@@ -252,13 +254,24 @@ func resourceLanDelete(ctx context.Context, d *schema.ResourceData, meta interfa
 		return diag.FromErr(err)
 	}
 
-	apiResponse, err := client.LANsApi.DatacentersLansDelete(ctx, dcId, d.Id()).Execute()
-	logApiRequestTime(apiResponse)
-
+	err := retry.RetryContext(ctx, d.Timeout(schema.TimeoutDelete), func() *retry.RetryError {
+		apiResponse, err := client.LANsApi.DatacentersLansDelete(ctx, dcId, d.Id()).Execute()
+		if err != nil {
+			if isDeleteProtected(apiResponse) {
+				log.Printf("[INFO] LAN %s is delete-protected, keep trying", d.Id())
+				return retry.RetryableError(fmt.Errorf("lan %s is delete-protected, keep trying %w", d.Id(), err))
+			}
+			if httpNotFound(apiResponse) {
+				log.Printf("[INFO] LAN already deleted %s", d.Id())
+				return nil
+			}
+			return retry.NonRetryableError(fmt.Errorf("an error occurred while deleting lan dcId %s ID %s %w", dcId, d.Id(), err))
+		}
+		log.Printf("[DEBUG] resource with id %s still has not been deleted", d.Id())
```

Comment:
@cristiGuranIonos I think that if we want to log a debug message here, it should be something like:
```suggestion
		log.Printf("[DEBUG] deletion started for LAN with ID: %v", d.Id())
```